estimate_length.py

```diff
@@ -0,0 +1,335 @@
+
+def prob_pAi_given_d(pAi, interval, read_coordinate, f, prob_f):
+    """Computes the conditional probability P(pAi|d) for a pAi to give 
+       rise to the read d. Prior probabilities for each pAi are taken to
+       be homogeneous, namely 1/N, N=number of pAis."""
+    nominator = prob_d_given_pAi(read_coordinate, pAi, interval, f, prob_f)
+    denominator = sum([prob_d_given_pAi(read_coordinate, pAi, intrv, f, prob_f) for intrv in range(len(pAi))])
+    return nominator/denominator
+
+def prob_d_given_L(read_coordinate, pAi, interval, Length, f, prob_f, length_range):
+    """Computes the conditional probability P(d|L) given the genomic coordinate
+       of the read, a set of pAis, which of the pAis is the polyA tail, a length
+       value, a bioanalyzer and a range for L."""
+    nominator = sum(prob_f * 1/Length * step_function(int(pAi[interval]['start']) + Length - read_coordinate - f))
+    norm_factor = sum([sum(prob_f * 1/length *
+                       step_function(int(pAi[interval]['start']) + length - read_coordinate - f)) for length in length_range])
+    return nominator/norm_factor
+
+def prob_d_given_L_weighted(read_coordinate, pAi, interval, Length, f, prob_f, length_range):
+    """Computes the conditional probability P(d|L) given the genomic coordinate
+       of the read, a set of pAis, which of the pAis is the polyA tail, a length
+       value, a bioanalyzer and a range for L."""
+    pAi[interval]['end'] = int(pAi[interval]['start']) + Length
+    nominator = sum(prob_f * 1/Length * step_function(int(pAi[interval]['end']) - read_coordinate - f) *
+                    prob_d_given_pAi(read_coordinate, pAi, interval, f, prob_f))
+
+    # compute the norm_factor for sum(prob)=1
+    norm_factor = 0
+    for length in length_range:
+        pAi[interval]['end'] = int(pAi[interval]['start']) + length
+        norm_factor += sum(prob_f * 1/length * step_function(int(pAi[interval]['end']) - read_coordinate - f) *
+                           prob_d_given_pAi(read_coordinate, pAi, interval, f, prob_f))
+    return nominator/norm_factor
+
+def estimate_poly_tail_length(reads, tail_range, pAi, interval, f, prob_f, weighted):
+    """Takes a set of reads (list of read_coordinates), a range of polyA tail
+       lengths, a set of internal priming intervals and a bioanalyzer profile.
+       Homogeneous prior probabilities for the p(L) are assumed."""
+    L_probs = []
+#    nominator = np.ones(len(tail_range))
+    nominator = np.zeros(len(tail_range))
+    for read in reads:
+        read_probs = []
+        for L in tail_range:
+            if weighted:
+                read_probs.append(prob_d_given_L_weighted(read, pAi, interval, L, f, prob_f, tail_range))
+            else:
+                read_probs.append(prob_d_given_L(read, pAi, interval, L, f, prob_f, tail_range))
+        for index in range(len(read_probs)):
+            if read_probs[index] == 0.0:
+                read_probs[index] = 0.00000000001
```

Comment:
Is this some sort of a pseudocount? Should we make this a parameter or at lease give it a name to make it more clear?